tests/h/services/group_list_test.py

```diff
@@ -380,16 +194,122 @@ def test_groups_are_sorted_alphabetically(
         assert [group.pubid for group in results] == expected_sorted_pubids
 
 
-class TestListGroupsFactory(object):
-    def test_list_groups_factory(self, pyramid_request):
-        svc = list_groups_factory(None, pyramid_request)
+class TestUserGroups(object):
+    def test_it_returns_all_of_a_users_groups(self, svc, user, user_groups):
+        user.groups = user_groups
+
+        u_groups = svc.user_groups(user)
+
+        for group in user.groups:
+            assert group in u_groups
+        assert len(u_groups) == len(user_groups)
+
+    def test_it_sorts_a_users_groups_by_name(self, svc, user, user_groups):
+        user.groups = user_groups
+
+        u_groups = svc.user_groups(user)
+
+        group_names = [group.name for group in u_groups]
```

Comment:
The first test is asserting that all of the correct groups are returned. The second is a test for ordering, as the name implies. Sure they share some setup, but...it's valid to say they're two separate tests, is it not?

Reply:
The way I look at it is, if something changed/was broken would the test catch it. Since the ordering check also checks to make sure the groups are returned, the ordering check would fail in both cases.  This means the first test isn't adding any additional checks that aren't covered by the second test thus the first test is unnecessary.

Reply:
Maybe test_it_returns_all_of_a_users_groups_sorted_by_group_name